x-pack/filebeat/input/awss3/config.go

```diff
@@ -78,6 +81,16 @@ func (c *config) Validate() error {
 		return fmt.Errorf("number_of_workers <%v> must be greater than 0", c.NumberOfWorkers)
 	}
 
+	if c.QueueURL != "" && c.RegionName != "" {
+		u, err := url.Parse(c.QueueURL)
+		if err != nil {
+			return fmt.Errorf("invalid queue_url: %w", err)
+		}
+		if strings.HasSuffix(u.Host, ".amazonaws.com") {
```

Comment:
There are other AWS domains besides just `amazonaws.com` that you should include.  See https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/awss3/input.go#L350